src/huggingface_hub/errors.py

```diff
@@ -84,6 +84,16 @@ def append_to_message(self, additional_message: str) -> None:
         """Append additional information to the `HfHubHTTPError` initial message."""
         self.args = (self.args[0] + additional_message,) + self.args[1:]
 
+    def __reduce_ex__(self, protocol):
+        """Fix pickling of Exception subclass with kwargs"""
+        args = (str(self),)
+        kwargs = {"response": self.response, "server_message": self.server_message}
+        return self._from_args, (args, kwargs)
+
+    @classmethod
+    def _from_args(cls, args, kwargs):
+        return cls(*args, **kwargs)
```

Comment:
```suggestion
    def __reduce_ex__(self, protocol):
        """Fix pickling of Exception subclass with kwargs"""
        return (
            self.__class__,
            (str(self),),
            {"response": self.response, "server_message": self.server_message}
        )
```

wouldn't this work to avoid defining `_from_args`?